tests/components/broadlink/test_heartbeat.py

```diff
@@ -0,0 +1,100 @@
+"""Tests for Broadlink heartbeats."""
+from unittest.mock import call, patch
+
+from homeassistant.components.broadlink.heartbeat import BroadlinkHeartbeat
+from homeassistant.util import dt
+
+from . import get_device
+
+from tests.common import async_fire_time_changed
+
+DEVICE_PING = "homeassistant.components.broadlink.heartbeat.blk.ping"
+
+
+async def test_heartbeat_trigger_startup(hass):
+    """Test that the heartbeat is initialized with the first config entry."""
+    device = get_device("Office")
+
+    with patch(DEVICE_PING) as mock_ping:
+        await device.setup_entry(hass)
+        await hass.async_block_till_done()
+
+    assert mock_ping.call_count == 1
+    assert mock_ping.call_args == call(device.host)
+
+
+async def test_heartbeat_ignore_oserror(hass):
+    """Test that an OSError is ignored."""
+    device = get_device("Office")
+
+    with patch(DEVICE_PING, side_effect=OSError()):
+        await device.setup_entry(hass)
+        await hass.async_block_till_done()
+
+
+async def test_heartbeat_trigger_right_time(hass):
+    """Test that the heartbeat is triggered at the right time."""
+    device = get_device("Office")
+
+    await device.setup_entry(hass)
+    await hass.async_block_till_done()
+
+    with patch(DEVICE_PING) as mock_ping:
+        async_fire_time_changed(hass, dt.utcnow() + BroadlinkHeartbeat.INTERVAL)
+        await hass.async_block_till_done()
+
+    assert mock_ping.call_count == 1
```

Comment:
Why not 2 ? you send a heartbeat direct hearthbeat.py#32 and here you add INTERVAL so it should have fired again.

Reply:
The call_count starts from the context manager. The setup was done before.

Reply:
you are right.